lib/IDE/CodeCompletion.cpp

```diff
@@ -1371,12 +1599,21 @@ MutableArrayRef<CodeCompletionResult *> CodeCompletionContext::takeResults() {
 Optional<unsigned> CodeCompletionString::getFirstTextChunkIndex(
     bool includeLeadingPunctuation) const {
   for (auto i : indices(getChunks())) {
-    auto &C = getChunks()[i];
+    const Chunk &C = getChunks()[i];
     switch (C.getKind()) {
     using ChunkKind = Chunk::ChunkKind;
     case ChunkKind::Text:
+      // Skip white-space only chunks.
+      if (C.getText().find_first_not_of(" \r\n") == StringRef::npos)
```

Comment:
Is it worth using Clang's `isWhitespace` which also handles "\t\f\v"?  Or can that never happen here?

Also, do we ever have completions whose only  text is a space? I'm hoping "no", but we used to not allow empty completions and then we ended up adding those at some point so I wanted to check.

Reply:
My initial implementation was `.ltrim().size() == 0` which also handles `\t\v\f`. But since this is a somewhat hot-path, I wanted to minimize the character set to check, and avoid the [overhead of `std::min` and `drop_front`](https://llvm.org/doxygen/StringRef_8h_source.html#l00849). I don't think we ever print `\t\v\f` in `ASTPrinter`.

Reply:
SGTM. For what it's worth the clang function ought to be quite fast as well once the lookup table is in L1 cache, and the implementation is completely inlinable, but I wouldn't know off hand which one is better here.  We can always change it later.